src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -2976,30 +2976,16 @@ internal bool LoadModuleManifestData(ExternalScriptInfo scriptInfo, ManifestProc
                     BaseCmdletPatterns = oldCmdletPatterns;
                 }
 
-                // If there is an existing session state and the new module info
-                // session state is empty, then use the existing session state.
-                // This will be the case when ModuleToProcess is a binary module
-                // and there were nested modules.
-                if (newManifestInfo.SessionState == null && ss != null)
-                {
-                    newManifestInfo.SessionState = ss;
-                    ss.Internal.Module = newManifestInfo;
-                }
-                // If the new module info session state is not empty but there is no
-                // existing session state, then use the new module info's sessions state;
-                // This will be the case if the module to process is a script module
-                // but there where no nested modules.
-                else if (newManifestInfo.SessionState != null && ss == null)
```

Comment:
`ss == null` only if we are doing module analysis, but when we are doing module analysis, `newManifestInfo.SessionState` is always `null`, because we don't really load anything.